src/modules/kvs/kvsroot.c

```diff
@@ -253,6 +254,44 @@ int kvsroot_mgr_iter_roots (kvsroot_mgr_t *krm, kvsroot_root_f cb, void *arg)
 /* Convenience functions on struct kvsroot
  */
 
+const char *kvsroot_save_transaction_request (struct kvsroot *root,
+                                              const flux_msg_t *request,
+                                              uint32_t rank,
+                                              unsigned int seq)
+{
+    char *name = NULL;
+
+    if (!root || !request) {
+        errno = EINVAL;
+        goto error;
+    }
+
+    if (asprintf (&name, "transaction_req.%u.%u", rank, seq) < 0)
+        goto error;
+
+    if (flux_msg_aux_set (request,
+                          "transaction_req_name",
+                          name,
+                          (flux_free_f) free) < 0)
+        goto error;
+
+    if (zhash_insert (root->transaction_requests,
+                      name,
+                      (void *)flux_msg_incref (request)) < 0) {
+        errno = EEXIST;
+        goto error;
+    }
```

Comment:
Just wondering if this is the best container?  I think we decided of the hashes, `zhashx` had more sane growth management than `zhash`.  I guess if the number of messages stored in the hash is always small that may not come into it.

Would a `flux_msglist_t` possibly be a better fit or is the lookup by name important?

Sorry I'm not entirely remembering how this code works so I'm just asking the question.

Reply:
I was thinking of that.  But left it as a `zhash` just because that was what was there before.  Could think of that for some potential cleanup / improvement.

I had contemplated a `zlist` b/c now that we only support commits, the "transactions" should reasonably be done in order.  So name matching should always occur with the first entry.  But not sure if that's worth it vs a hash since we would want to double check and match names anyways.